src/sentry/models/widget.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import, print_function
+
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.constants import ObjectStatus
+from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+
+
+class WidgetDisplayTypes(object):
+    BASIC_LINE_CHART = 1
+    TWO_QUERY_LINE_CHART = 2
+    MAP = 3
+    HORIZONATAL_BAR_CHART = 4
+    TIMELINE = 5
+    STACKED_AREA = 6
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.BASIC_LINE_CHART, 'basic_line_chart'),
+            (cls.TWO_QUERY_LINE_CHART, 'two_query_line_chart'),
+            (cls.MAP, 'map'),
+            (cls.HORIZONATAL_BAR_CHART, 'horizontal_bar_chart'),
+            (cls.TIMELINE, 'timeline'),
+            (cls.STACKED_AREA, 'stacked_area')
+        ]
+
+
+class WidgetDataSourceTypes(object):
+    DISCOVER_SAVED_SEARCH = 1
+    API_QUERY = 2  # ehhhh.....
```

Comment:
This seems rather broad. But I thought it might be a fine placeholder so you can see what I'm thinking.